hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/scope/HandleHederaNativeOperations.java

```diff
@@ -117,14 +116,15 @@ public void setNonce(final long contractNumber, final long nonce) {
      * {@inheritDoc}
      */
     @Override
-    public void finalizeHollowAccountAsContract(@NonNull final Bytes evmAddress) {
+    public AccountID finalizeHollowAccountAsContract(@NonNull final Bytes evmAddress, final long contractNonce) {
         requireNonNull(evmAddress);
         final var accountStore = context.readableStore(ReadableAccountStore.class);
         final var hollowAccountId = requireNonNull(accountStore.getAccountIDByAlias(evmAddress));
         final var tokenServiceApi = context.serviceApi(TokenServiceApi.class);
-        tokenServiceApi.finalizeHollowAccountAsContract(hollowAccountId, INITIAL_CONTRACT_NONCE);
+        tokenServiceApi.finalizeHollowAccountAsContract(hollowAccountId, contractNonce);
```

Comment:
Just like the redundancy in `mono-service`, afaict this is a no-op, simply setting the existing hollow account's `nonce` to be what it already is.

To confirm, I just changed two lines from `develop`:
  1. Added `pendingCreation = null` before L#288 [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/ProxyWorldUpdater.java#L288).
  2. Commented out [this line](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/api/TokenServiceApiImpl.java#L175) that was always resetting the hollow account's nonce to `INITIAL_CONTRACT_NONCE`.
  
...and `canMergeCreate2MultipleCreatesWithHollowAccount()` passed with no other changes. 😁 

So I think we can just do that `pendingCreation = null` trick and remove the `contractNonce` parameter from ` TokenServiceApi.finalizeHollowAccountAsContract()` and this issue will be fixed.

Reply:
addressed. @MiroslavGatsanoga thank you for the guiding questing. @tinker-michaelj thank you for the suggestion. This way the solution is way more elegant. 🙏